backend/packages/Upgrade/src/api/services/QueryService.ts

```diff
@@ -34,13 +37,35 @@ export class QueryService {
 
     const promiseResult = await Promise.all(promiseArray);
     const analysePromise = promiseResult.map((query) => this.logRepository.analysis(query));
-    let response = await Promise.all(analysePromise);
-    response = response.map((res, index) => {
-      return {
-        id: queryIds[index],
-        result: res,
-      };
+    const response = await Promise.allSettled(analysePromise);
+
+    const failedQuery = Array<Promise<any>>();
+
+    let modifiedResponse = response.map((query, index) => {
+      if (query.status === 'fulfilled') {
+        return query.value;
+      } else {
+        this.log.error('Error in Query Id ', queryIds[index]);
+        failedQuery.push(this.errorService.create({
+          endPoint: '/api/query/analyse',
+          errorCode: 500,
+          message: `Query Failed error: ${JSON.stringify(queryIds[index], undefined, 2)}`,
+          name: 'Query Failed error',
+          type: SERVER_ERROR.QUERY_FAILED,
+        } as any));
```

Comment:
Can you please update the `any` with `ExperimentError`?

Reply:
Thanks, code updated